Instruments/source/DYFileMerger.cxx

```diff
@@ -0,0 +1,141 @@
+            CalculateWeight(output_bin);
+        }
+        DYBinDescriptor::SaveCfg(args.output_file(), output_bins);
+    }
+
+
+private:
+    SampleDescriptor<DYBinDescriptor, ntuple::GenEventCountMap> global_map;
+    SampleDescriptor<DYBinDescriptor, ntuple::GenEventCountMap> inclusive;
+    VectorSampleDescriptor all_samples;
+    VectorDYBinDescriptor output_bins;
+    Arguments args;
+
+
+    void LoadInputs()
+    {
+        analysis::ConfigReader config_reader;
+
+        DYBinDescriptorCollection file_descriptors;
+        DYFileConfigEntryReader file_entry_reader(file_descriptors);
+        config_reader.AddEntryReader("FILE", file_entry_reader, true);
+
+        config_reader.ReadConfig(args.file_cfg_name());
+
+        for (auto file_descriptor : file_descriptors){ //loop on DYJets files
+            const DYBinDescriptor file_descriptor_element = file_descriptor.second;
+            std::cout << "File descriptor characteristics: " << file_descriptor.first << ", " <<
+                         file_descriptor_element.file_path << ", " << file_descriptor_element.fileType
+                      << std::endl;
+            SampleDescriptor<DYBinDescriptor, ntuple::GenEventCountMap> sample_desc;
+            sample_desc.bin = file_descriptor_element;
+            //global_map.bin = file_descriptor_element;
+            if (file_descriptor_element.fileType == FileType::inclusive)
+                inclusive.bin = file_descriptor_element;
+
+            std::string filename = args.input_path() + file_descriptor_element.file_path;
+            auto inputFile = root_ext::OpenRootFile(filename);
+            ntuple::SummaryTuple summaryTuple(args.tree_name(), inputFile.get(), true);
+            const Long64_t n_entries = summaryTuple.GetEntries();
+
+            for(Long64_t current_entry = 0; current_entry < n_entries; ++current_entry) { //loop on entries
+                summaryTuple.GetEntry(current_entry);
+                for (size_t i = 0; i < summaryTuple.data().lhe_n_partons.size(); ++i){ // loop on gen event info
+                    const ntuple::GenId genId(summaryTuple.data().lhe_n_partons.at(i),
+                                              summaryTuple.data().lhe_n_b_partons.at(i),
+                                              summaryTuple.data().lhe_ht10_bin.at(i));
+                    size_t nevents = summaryTuple.data().lhe_n_events.at(i);
+                    sample_desc.gen_counts[genId] += nevents;
+                    global_map.gen_counts[genId] += nevents;
+                    if (file_descriptor_element.fileType == FileType::inclusive)
+                        inclusive.gen_counts[genId] += nevents;
+
+                } // end loop on gen event info
+            } //end loop on entries
+            all_samples.push_back(sample_desc);
+        } //end loop n file_descriptors
+    }
+
+
+    void CalculateWeight(DYBinDescriptor& output_bin) const
+    {
+        size_t all_events = global_map.Integral(output_bin);
```

Comment:
Sorry, missed this one: you can also declare it as double. this will allow to remove remaining double(...) and avoid possible problems with std::pow